aries_vcx/src/new_protocols/mod.rs

```diff
@@ -0,0 +1,425 @@
+use aries_vcx_core::{
+    anoncreds::base_anoncreds::BaseAnonCreds,
+    ledger::base_ledger::{AnoncredsLedgerRead, AnoncredsLedgerWrite, IndyLedgerRead, IndyLedgerWrite},
+    wallet::base_wallet::BaseWallet,
+};
+use async_trait::async_trait;
+use chrono::Utc;
+use diddoc_legacy::aries::diddoc::AriesDidDoc;
+use messages::{
+    decorators::{thread::Thread, timing::Timing},
+    msg_fields::protocols::connection::{
+        invitation::{
+            InvitationContent, PairwiseDidInvitationContent, PairwiseInvitationContent, PublicInvitationContent,
+        },
+        request::RequestContent,
+        response::ResponseContent,
+        ConnectionData,
+    },
+};
+use url::Url;
+
+use crate::{
+    common::{
+        ledger::transactions::get_service,
+        signing::{decode_signed_connection_response, sign_connection_response},
+    },
+    errors::error::VcxResult,
+};
+
+pub struct BootstrapInfo {
+    service_endpoint: Url,
+    recipient_keys: Vec<String>,
+    routing_keys: Vec<String>,
+    did: Option<String>,
+    service_endpoint_did: Option<String>,
+}
+
+#[async_trait]
+pub trait MessageHandler {
```

Comment:
This MessageHandler approach, while it makes it more convenient to write the implementations of the functions inside, suffers from similar weaknesses as our (recently remove) approach of passing in `Arc<Profile>` in function arguments.

- Breaking law of Demeter.  In order to process a message, I need to construct handler which needs to have set up `LedgerRead`, `LedgerWrite`, ... etc. while in practice, the function I am about to call might be only using some or none of the components. 
Then that leads to passing in mock implementations "when you know ahead of time the function you are planning to call will not internally leverage that particular component".
- It's also remove hits about how certain functions work. For example with old style functions, you would have something like `process_request(read: Arc<dyn LedgerRead>, msg: Message)` so you can tell at glance - "ah, this function will be reading ledger"

Reply:
I don't have a strong opinion on this. It was indeed mainly a left over that was adjusted to make function signatures shorter.

Reply:
Also, this kind of helped with the test and creating the two entities, but you already know I'm not in favor of testing dictating production code 😅 .

Another possible use case would be uniform decorator handling, but idk if this really should reside in here or not. The argument about having to pass a lot of components which might not even be used is good though, so I'll refactor this to only be used for entities in the test.

Reply:
`MessageHandler` was officially demoted 😄